studio/components/layouts/ProjectLayout/NavigationBar/NavigationBar.utils.tsx

```diff
@@ -33,23 +34,21 @@ export const generateProductRoutes = (ref: string, project?: ProjectBase): Route
         />
       ),
       link: isProjectBuilding ? buildingUrl : `/project/${ref}/editor`,
+      hidden: !usePermissions(PermissionAction.SQL_SELECT, 'postgres.public'),
```

Comment:
i don't think we hide links when a user has no permission.
we should have an empty state on the screen, explaining they do not have permission to use this.

Reply:
Should there be a different UI treatment for the menu icons or labels where a user doesn't have permissions to access then? The icons and labels are already rather grey so changing the colour might not be obvious. What about appending a `(no access)` to the labels, or prepending a cross unicode symbol or emoji?

Reply:
There are also possibly cases where some screens will have some read only parts, and then some read/write parts

Reply:
also need to consider self hosting, and what impact using this hook will have.

Reply:
For now, I had included `if (!IS_PLATFORM) return true` for `usePermissions()` so that permissions wouldn't apply when self hosting.